File: yunu/프로그래머스/인사고과/좋은 코드.js
function solution(scores) {
  const wanhoScore = scores[0];
  let currMaxScore = 0;
  let answer = 1;

  scores.sort((a, b) => (a[0] === b[0] ? a[1] - b[1] : b[0] - a[0]));

  for (let i = 0; i < scores.length; i++) {
    if (scores[i][1] < currMaxScore) {
      if (scores[i] == wanhoScore) return -1;
    } else {
      currMaxScore = Math.max(scores[i][1], currMaxScore);
      if (scores[i][0] + scores[i][1] > wanhoScore[0] + wanhoScore[1])
        answer += 1;
    }
  }

  return answer;
}

console.log(
  solution([
    [2, 2],
    [1, 4],
    [3, 2],
    [3, 2],
    [2, 1],
  ])
);

console.log(
  solution([
    [5, 1],
    [5, 0],
    [4, 1],
    [3, 2],
    [2, 3],
    [1, 4],
    [0, 5],
  ])
);

// console.log(solution([[0, 1]]));

File: yunu/프로그래머스/스타 수열/좋은 코드.py
def solution(a):
    dic = {}
    check = {}
    for i in range(len(a)):
        dic[i] = 0
        check[i] = -2

    for i in range(len(a) - 1):
        if a[i] != a[i+1]:
            if check[a[i]] != i-1:
                dic[a[i]] += 1
                check[a[i]] = i
            if check[a[i+1]] != i-1:
                dic[a[i+1]] += 1
                check[a[i+1]] = i

    return max(dic.values()) * 2

# print(solution([0]))
print(solution([5, 2, 3, 3, 5, 3]))
print(solution([0, 3, 3, 0, 7, 2, 0, 2, 2, 0]))